### DISCUSSION TITLE
Setting workflow environment variables with the GitHub Actions Toolkit

### DISCUSSION THREAD

**Discussion by cbellofx** [community/community#25166] [Actions]
<p>Hello.</p>
<p>I am currently creating a javascript action and leveraging the Github Actions Toolkit and would like to be able to set Environment variables for an entire workflow.</p>
<p>There is a function called exportVariable as part of the <span class="mention">@action</span>/core package that seems to fit the bill but only seems to set environment variables at the job level.</p>
<p>Is there a programmatic way to modify workflow level environment variables?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/cbellofx">@cbellofx</a> ,</p>
<p>It’s not supported to modify workflow level env variables from an javascript/docker action. And currently in workflow file, workflow level env don’t accept variables.</p>
<p>Thanks.</p>

**Comment by cbellofx** [comment]
<p>Thanks <a class="mention" href="/weide-zhou">@weide-zhou</a> </p>
